### PR TITLE
Google Tag Manager integration guide

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
A guide on implementing front-end GB integration via Google Tag Manager.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2695,6 +2695,13 @@
     uuid "^8.0.0"
     xdg-basedir "^4.0.0"
 
+"@growthbook/growthbook-react@^0.12.0":
+  version "0.12.0"
+  resolved "https://registry.yarnpkg.com/@growthbook/growthbook-react/-/growthbook-react-0.12.0.tgz#c97d40d3cc22b56d04e170d9d7149ea1567fac6f"
+  integrity sha512-Oo2cbGpM8thaAmwEccKFy/uRU3bqU4r4xcpKaLEHPeGARCkccFi1tdq2GPYEoO6cyeo7/Wc9eHqhtazg340orA==
+  dependencies:
+    "@growthbook/growthbook" "^0.23.0"
```
Comment (tinahollygb) [inline yarn.lock:2703]: These changes seem unrelated to your documentation-related changes and are likely to create conflicts with other PR's that change dependencies.

Comment (bryce-fitzsimons) [reply yarn.lock]: oops

File: docs/docs/guide/google-tag-manager-and-growthbook.mdx [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+description: This guide walks you through using GrowthBook with Google Tag Manager (GTM) to track your experiments and measure their impact on your business.
+sidebar_label: Google Tag Manager (GTM)
+slug: google-tag-manager-and-growthbook
+---
+
+# Using GrowthBook with Google Tag Manager (GTM)
+
+Now customers who are familiar with feature management using Google Tag Manager (GTM), yet may lack the engineering resources or capability to implement changes in their codebase, may easily use GrowthBook with GTM to power their AB tests. This setup is commonly used by marketing teams and CRO agencies.
+
+Note: GrowthBook also offers a [visual editor](https://docs.growthbook.io/app/visual) for lightweight, no-code UI changes.
```
Comment (Auz) [inline docs/docs/guide/google-tag-manager-and-growthbook.mdx:12]: Not sure if you want to, but we have the 
:::note 
some text
:::
components for adding callouts.

File: docs/docs/guide/google-tag-manager-and-growthbook.mdx [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+In this guide, we will assume familiarity with the GTM platform. We also assume the ability to apply a unique user ID via Google Analytics, cookies, etc; as well as the ability to track events (GA or otherwise).
```
Comment (Auz) [inline docs/docs/guide/google-tag-manager-and-growthbook.mdx:14]: Can you use GA4 clientId that we explain a few sections lower down?

File: docs/docs/guide/google-tag-manager-and-growthbook.mdx [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+The second strategy does not incur as much of a page render speed tax - the flicker is much less noticeable. However, you’ll need to hard-code all of your experiments and their rulesets, as well as any feature flag rules and overrides, into the GTM tag’s source code. Each time you need to change your experiments or phases, you’ll need to modify the code snippet and publish a new GTM version. For some, this may impose a significant burden; for others, the render speed improvement may be worth the tax.
+
+If neither of these strategies seem ideal, we’d encourage you to pursue a traditional GrowthBook implementation without GTM, preferably with back-end or hybrid feature flag evaluation.
```
Comment (Auz) [inline docs/docs/guide/google-tag-manager-and-growthbook.mdx:24]: Can we link anywhere for these users to learn more?

File: docs/docs/guide/google-tag-manager-and-growthbook.mdx [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+<script id="growthbook-sdk" src="https://cdn.jsdelivr.net/npm/@growthbook/growthbook/dist/bundles/index.min.js" defer></script>
+```
+
+To publish the SDK tag, submit our workspace changes (the blue “Submit” button on the top of the GTM application), Then ensure “Publish and Create version” is selected and click the blue “Publish” button – or use whichever GTM release strategy you are already using.
+
+## Initializing the SDK
+
+To actually use the GrowthBook SDK on our pages to control on-page features with AB tests, we’ll need to create another code snippet and load it through GTM. While you can certainly add the snippet to your existing tag (the GrowthBook SDK tag we created above), you may find it cleaner to add another Custom HTML tag and give it a name such as “GrowthBook Implementation” Be sure to set the firing triggers in the same way as you did above.
+
+Insert the following code into your tag:
```
Comment (Auz) [inline docs/docs/guide/google-tag-manager-and-growthbook.mdx:46]: Do we have to call out that they should replace the clickKey and other attributes? What about the tracking callback? - ah, read lower down - so perhaps we should let folks know we are going into detail and updating this snippet.

File: docs/docs/guide/google-tag-manager-and-growthbook.mdx [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+      // 3-way inline test
+      var result2 = gb.run({
+        key: "banner-experiment",
+        variations: ["A", "B", "C"],
+        weights: [0.5, 0.25, 0.25],
+        coverage: 1.0
+      });
+      if (result2.value === "variation-1") {
+        document.querySelector("banner1").innerHTML = "Click here in the next 24 hours";
+      } else if (res.value === "variation-2") {
```
Comment (jdorn) [inline docs/docs/guide/google-tag-manager-and-growthbook.mdx:340]: A couple typos here.
1. `res.value` should be `result2.value`
2. The values should be `B` and `C`, not `variation-1` and `variation-2`

File: docs/docs/guide/google-tag-manager-and-growthbook.mdx [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+
+# Using GrowthBook with Google Tag Manager (GTM)
+
+Now customers who are familiar with feature management using Google Tag Manager (GTM), yet may lack the engineering resources or capability to implement changes in their codebase, may easily use GrowthBook with GTM to power their AB tests. This setup is commonly used by marketing teams and CRO agencies.
+
+Note: GrowthBook also offers a [visual editor](https://docs.growthbook.io/app/visual) for lightweight, no-code UI changes.
+
+In this guide, we will assume familiarity with the GTM platform. We also assume the ability to apply a unique user ID via Google Analytics, cookies, etc; as well as the ability to track events (GA or otherwise).
+
+## Choosing a strategy: scalable and easy versus page speed
```
Comment (jdorn) [inline docs/docs/guide/google-tag-manager-and-growthbook.mdx:16]: I think in the majority of use cases, people will want the 2nd approach (inline experiments).  The wording in this guide makes it seem like it's really hard only and worth it if you need extreme performance. 

In reality, with GTM, inline experiments are not really any harder to implement (in fact it's often easier since you don't need to deal with API keys or configuring feature flag rules).  With both approaches, you need to deploy a new GTM version every time you create a new experiment.  The only difference is with feature flags, you can make changes later without another deploy.

I would at the very least, make the language less harsh in this section.  I would maybe even consider swapping the order and making Inline Experiments the default choice since I think it fits better with more people's use cases.

File: docs/docs/guide/google-tag-manager-and-growthbook.mdx (hunk 2 above)
Comment (bryce-fitzsimons) [reply docs/docs/guide/google-tag-manager-and-growthbook.mdx]: TIL, thanks!

**Review by jdorn — APPROVED** [review]
